### ISSUE TITLE
504 error when accessing WorldAnvil.com, cannot interact with site at all

### ISSUE THREAD

**Issue by LexaMichaelides** [worldanvil/worldanvil-bug-tracker#1278]
**World Anvil Username**: lexaMichaelides

**World / Campaign / RPG Character / Manuscript Title**: Taethir / Morning Glory

**Feature**: Campaign Sessions

**Operating System**: macOS Monterey 12.4

**Browser**: Firefox, Safari, Google Chrome

**Device / Device Model**: Mac Mini (M1), Macbook Pro (M2), iPhone 13 mini

**Related URLS** (To public & visible pages, add screenshots otherwise):
I can't actually access the site to get screenshots. 
![image](https://user-images.githubusercontent.com/32111123/191310606-e8e99cfa-a677-4002-876f-a957117b09c8.png)

**Describe the Issue**
The issue started mostly when accessing the campaign and session section of the site (namely https://www.worldanvil.com/heroes/campaign/session/list). The loading process stops and I get a blank white page or a 504 error.  It has been intermittent for about a week and is duplicated by my fellow world owner, JaeAlexisLee. 

After a week of this happening, I'm now completely unable to access the WorldAnvil site except for a few seconds immediately after clearing my cache.

I’m wondering if my campaign just holds too much data to be loading properly? I think I have several hundred session reports.

**To Reproduce**
1. Navigate to https://www.worldanvil.com/ 
2. Fail due to 504 error.

**Expected behavior**
I expected to be able to access the site and navigate as usual.

**Screenshots**
If applicable, add screenshots to help explain your problem. You can add screenshots by drag & drop into this window!

**Comment by iamromeo** [comment]
Hello @LexaMichaelides this might be related to the size of  your world - can you please let me know how many articles this world has and if there are any specific pages that cause this more than others?

**Comment by LexaMichaelides** [comment]
I would love to let you know, but unfortunately I can't access the site so I can't get that information. 

The campaign/sessions page caused the most problems and that is where the vast majority of my articles are.  I think I have around 250 session reports and they are all linearly linked in order. all of them are also nested under two categories.

**Comment by LexaMichaelides** [comment]
This would normally be the link to the homepage, I think: https://www.worldanvil.com/w/taethir-lexamichaelides
A friend who follows my world says he can't access it either anymore.

**Comment by iamromeo** [comment]
@LexaMichaelides that might be the issue that crashes your world. You should not have over 20 articles per category - you understand how 100s can be an issue - 

The  good thing is that I can see your homepage as it stands - Can you please send me an email at dimitris@worldanvil.com with your username and password (I would normally ask you to change it but if you cannot access it this might be a problem) so I can investigate your account?

**Comment by LexaMichaelides** [comment]
Yes, I'll email you. By 'nested under two categories' I mean that generally one child category holds, at most, 10 articles and then there is a parent category that holds maybe 5 child categories. I've done my best to make the filesystem sensible given the constraints.

**Comment by iamromeo** [comment]
Potentially related to #988

**Comment by LexaMichaelides** [comment]
@iamromeo checking in on this as it makes the entire site unusable for me. I understand that it may not be urgent from a product perspective but it matters to me, since if I can't access the site to even reorganize my world, I'll just lose all of my content.

**Comment by iamromeo** [comment]
@LexaMichaelides I am sorry I should have contacted you earlier - We are already investigating
- It is an issue specifically with your campaign section. 
- If you do not access it the slowdowns should not occur (they only begin once you do try to get into it because  the CDN throttles your connection) 
- I am stilll working on it and I have time allocated on it this week as well.

**Comment by LexaMichaelides** [comment]
I 100% believe that the data is still there and as a last resort I could request that an employee send it to me, but I still cannot access the site reliably even after clearing my cache and not clicking the campaign section. The site remains loading on the Dashboard for several minutes.

For now, the site remains unusable for how I organize my world since I only use the campaign section regularly! How exciting to be an edge case :) 

Thanks for working on it!

**Comment by LexaMichaelides** [comment]
@iamromeo checking in again. I'm trying to add content outside the campaign section but I still get timeouts.

**Comment by LexaMichaelides** [comment]
@iamromeo checking again.

**Comment by iamromeo** [comment]
hey @LexaMichaelides I must admit this is one of the hardest to replicate so far - I have been working on it for 2 weeks now. 
It seems that it is related with the daisy change of previous/next connected articles but I am trying to fix it without breaking your chain ... or anyone else's for that fact. 

First of all suffice it to say for all the time I will spend fixing the issue that you cannot really work with your world I will add time to your membership - double that 

Secondly - I am really on it and I will have something for you as soon as possible.

**Comment by LexaMichaelides** [comment]
@iamromeo, I understand. I'm no expert but I also work in software dev and have taught database courses and I don't envy your job of fixing whatever data structure issue this might be. 

The membership length is not so much an issue for me, but I'm wondering if there is some hacky alternative solution that would let you copy my world as-is and just break the daisy chain in my world so I can patch it on the user end and keep working using the campaign section. I don't mind breaking the linear linking of the world if all the hierarchical linking is intact. Or finding some other solution like only using linear linking in small batches (not like the 250 link chain monstrosity I have).

I run this game twice a week so even being able to add content to the world while you're working on the issue would help. I'm really not looking forward to the day when I have to copy a bunch of files into the system manually!

**Comment by iamromeo** [comment]
Give me until Wednesday to try to find a solution - I have some things in mind that I still want to try. 
If that fails we can talk alternatives asap.
